srml/support/procedural/src/storage/transfo.rs

```diff
@@ -0,0 +1,619 @@
+				#[serde(deny_unknown_fields)]
+				pub struct GenesisConfig#fparam {
+					#ph_field
+					#( #config_field )*
+					#( #genesis_extrafields )*
+				}
+
+				#[cfg(feature = "std")]
+				impl#fparam Default for GenesisConfig#sparam {
+					fn default() -> Self {
+						GenesisConfig {
+							#ph_default
+							#( #config_field_default )*
+							#( #genesis_extrafields_default )*
+						}
+					}
+				}
+
+				#[cfg(feature = "std")]
+				impl#fparam #scrate::runtime_primitives::BuildStorage for GenesisConfig#sparam {
+
+					fn build_storage(self) -> ::std::result::Result<(#scrate::runtime_primitives::StorageMap, #scrate::runtime_primitives::ChildrenStorageMap), String> {
+						let mut r: #scrate::runtime_primitives::StorageMap = Default::default();
+						let mut c: #scrate::runtime_primitives::ChildrenStorageMap = Default::default();
+
+						#( #builders )*
+
+						#scall(&mut r, &mut c, &self);
+
+						Ok((r, c))
+					}
+				}
+			}
+		} else {
+			quote!()
+		}
+}
+
+fn decl_storage_items(
+	scrate: &proc_macro2::TokenStream,
+	traitinstance: &Ident,
+	traittype: &syn::TypeParamBound,
+	cratename: &Ident,
+	storage_lines: &ext::Punctuated<DeclStorageLine, Token![;]>,
+) -> proc_macro2::TokenStream {
+
+	let mut impls = Vec::new();
+	for sline in storage_lines.inner.iter() {
+		let DeclStorageLine {
+			name,
+			storage_type,
+			default_value,
+			visibility,
+			..
+		} = sline;
+
+		let (is_simple, extracted_opt, stk, gettype) = match storage_type {
+			DeclStorageType::Simple(ref st) => (true, ext::extract_type_option(st), None, st),
+			DeclStorageType::Map(ref map) => (false, ext::extract_type_option(&map.value), Some(&map.key), &map.value),
+		};
+		let is_option = extracted_opt.is_some();
+		let fielddefault = default_value.inner.get(0).as_ref().map(|d|&d.expr).map(|d|quote!( #d ))
+			.unwrap_or_else(|| quote!{ Default::default() });
+
+		let typ = extracted_opt.unwrap_or(quote!( #gettype ));
+
+		let option_simple_1 = if !is_option {
+			// raw type case
+			quote!( unwrap_or_else )
+		} else {
+			// Option<> type case
+			quote!( or_else )
+		};
+		let implementation = if is_simple {
+			let mutate_impl = if !is_option {
+				quote!{
+					<Self as #scrate::storage::generator::StorageValue<#typ>>::put(&val, storage)
+				}
+			} else {
+				quote!{
+					match val {
+						Some(ref val) => <Self as #scrate::storage::generator::StorageValue<#typ>>::put(&val, storage),
+						None => <Self as #scrate::storage::generator::StorageValue<#typ>>::kill(storage),
+					}
+				}
+			};
+
+			// generator for value
+			quote!{
+
+				#visibility struct #name<#traitinstance: #traittype>(#scrate::storage::generator::PhantomData<#traitinstance>);
+
+				impl<#traitinstance: #traittype> #scrate::storage::generator::StorageValue<#typ> for #name<#traitinstance> {
+					type Query = #gettype;
+
+					/// Get the storage key.
+					fn key() -> &'static [u8] {
+						stringify!(#cratename #name).as_bytes()
```

Comment:
`Ident::to_string` should work here as well and the you don't need `as_bytes`.
Just `b"string"`